Lib/bdb.py

```diff
@@ -273,7 +292,21 @@ def user_exception(self, frame, exc_info):
         """Called when we stop on an exception."""
         pass
 
-    def _set_stopinfo(self, stopframe, returnframe, stoplineno=0):
+    def user_opcode(self, frame):
+        """Called when we are about to execute an opcode."""
+        pass
+
+    def _set_trace_opcodes(self, trace_opcodes):
+        if trace_opcodes != self.trace_opcodes:
+            self.trace_opcodes = trace_opcodes
+            frame = self.__curframe
```

Comment:
Yes the reason for the double-underscore is to avoid the conflict with `pdb.Pdb.curframe`. And yes, we only want to set `trace_opcodes` in the "user frames", which means frames above (including) the caller frame into the pdb. So we need to record the entering frame.

The `curframe` attribute of `pdb` means the current frame being debugged. Do you prefer to change the the name `__curframe` to something like `entering_frame`?

Reply:
Yeah, that feels a lot less cryptic and scary. :) Especially considering we already have `stopframe`, `returnframe`, and `botframe`.